src/facade/dragonfly_connection.cc

```diff
@@ -297,7 +299,15 @@ void Connection::UnregisterShutdownHook(ShutdownHandle id) {
 
 bool Connection::IsReplicaCommand(util::tls::TlsSocket::Buffer buff) const {
   string_view str{reinterpret_cast<const char*>(buff.data()), buff.size()};
-  return str.starts_with("PING") || str.starts_with("DFLY") || str.starts_with("REPLCONF");
+  auto starts_with = [](const string_view source, const string_view starter) {
+    if (source.size() < starter.size()) {
+      return false;
+    }
+    size_t pos = 0;
+    return std::all_of(source.begin(), source.begin() + starter.size(),
+                       [&pos, starter](const char c) mutable { return c == starter[pos++]; });
```

Comment:
I'd replace these lines with:
```cpp
return source.substr(0, starter.size()) == starter);
```

Reply:
Or better yet: use `absl::StartsWith()` (https://github.com/abseil/abseil-cpp/blob/master/absl/strings/match.h#L58)

Reply:
I was expecting this comment here since you mentioned absl with `StartsWith`  on my other PR